apis/v1beta1/shared_types.go

```diff
@@ -75,6 +85,10 @@ type ParentReference struct {
 	// are specified, the name and port of the selected listener must match
 	// both specified values.
 	//
+	// * Service: Named port. When both Port (experimental) and SectionName
+	// are specified, the name and port of the selected port must match
+	// both specified values.
+	//
```

Comment:
@howardjohn I wasn't sure if _named_ port support made it into any upstream conformance yet, or if that was just an Istio-specific implementation. Adding this feels reasonable to me, although I think it should be Extended conformance for the mesh profile, to align with how Port is supported (rather than Core as when used to target a Gateway Listener).

Reply:
Ah, I may have just been inferring more functionality than actually existed from your comments in https://github.com/kubernetes-sigs/gateway-api/pull/1995